heaps/min_heap.py

```diff
@@ -21,18 +21,50 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            
+            Time Complexity: Adding an element to the heap is o(n) for the actual addition of the element to the end of the array.
+            It is O(log n) for the heap_up process as it traverses the heap so overall it is a O(log n) operation. 
+    
+            Space Complexity: The space complexity is constant O(n) because only one new node is added at a time.
```

Comment:
👍  The space complexity here is O(1) because you only add 1 element to the heap each time this function is called, and you have an iterative solution instead of a recursive one.